### PR TITLE
feat: allow configuring an engine context manager

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

This PR introduces a configurable context manager that wraps the `create_engine` call to analytical databases. The intended use case is to configure the Oracle driver with tenant-specific settings in a multi-tenant environment, but there are many other potential use cases for this:

1. Custom logging
2. Supressing noisy logs from 3rd party database drivers
3. `chroot`ing to prevent 3rd party databases drivers from accessing the filesystem (maybe using [pychroot](https://pypi.org/project/pychroot/))

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

Added a unit test.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -1142,16 +1143,18 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(  # pylint: disable=invalid-name
 # uploading CSVs will be stored.
 UPLOADED_CSV_HIVE_NAMESPACE: str | None = None
 
+
 # Function that computes the allowed schemas for the CSV uploads.
 # Allowed schemas will be a union of schemas_allowed_for_file_upload
 # db configuration and a result of this function.
+def allowed_schemas_for_csv_upload(  # pylint: disable=unused-argument
```
Comment (betodealmeida) [inline superset/config.py:1150]: I rewrote this as a function since it's easier to read and removes the lambda complaint from the linter.

**Review by mistercrunch — APPROVED** [review]
Nice. Did grepping to check if there were some rogue `create_engine` calls not going through this method and it looked squeaky cleaned.

**Review by dpgaspar — APPROVED** [review]
Nice!

**Comment by betodealmeida** [comment]
> Nice. Did grepping to check if there were some rogue `create_engine` calls not going through this method and it looked squeaky cleaned.

@mistercrunch yeah, when we implemented SSH tunneling we had to consolidate all the logic in a single place.

**Pull request merged**